packages/core/echo/echo-db/src/testing/echo-test-builder.ts

```diff
@@ -138,8 +139,13 @@ export const createDataAssertion = ({
       await db.flush();
     },
     verify: async (db: EchoDatabase) => {
-      const { objects } = await db.query().run();
-      const received = objects.find((object) => object.id === seedObject.id);
+      const findReceivedObject = async () => {
+        const { objects } = await db.query().run();
+        const received = objects.find((object) => object.id === seedObject.id);
+        return { objects, received };
+      };
+      await waitForCondition({ condition: async () => (await findReceivedObject()).received != null });
```

Comment:
hmm, doesn't this PR fix this exact issue? queries not returning all of the objects on the initial call